src/types/bone.d.ts

```diff
@@ -0,0 +1,410 @@
+import DataTypes, { AbstractDataType, DataType } from "../data_types";
+import { 
+  Pool, Literal, WhereConditions,
+  Collection, ResultSet, InstanceValues, OrderOptions,
+  QueryOptions, AttributeMeta, AssociateOptions, Values, Connection,
+} from './common';
+import { AbstractDriver } from '../drivers';
+import { Spell } from '../spell';
+
+export type RawQueryResult = typeof AbstractBone | ResultSet | boolean | number;
+
+interface SyncOptions {
+  force?: boolean;
+  alter?: boolean;
+}
+
+interface TransactionOptions {
+  connection: Connection;
+}
+
+export class AbstractBone {
```

Comment:
`static find` `static findOne` `static update` and others overloaded methods  cannot be override by subclass definitions because of different args required, so create a `AbstractBone` as superclass of `Bone` and `SequelizeBone`

Reply:
👌

should we put this dts file along side with src/bone.js like the others (src/spell.d.ts, src/adapters/sequelize.d.ts) as well?

Reply:
done